src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -512,16 +513,32 @@ public FormValidation doTestConnection(@QueryParameter String name, @QueryParame
                         Util.fixEmpty(serverCertificate), Util.fixEmpty(credentialsId), skipTlsVerify,
                         connectionTimeout, readTimeout).createClient();
 
+                // test listing pods
                 client.pods().list();
-                return FormValidation.ok("Connection successful");
+
+                // test creating pods
+                Pod pod = client.pods().create(new PodBuilder() //
+                        .withNewMetadata().withGenerateName("kubernetes-plugin-").endMetadata() //
+                        .withNewSpec() //
+                        .withContainers( //
+                                new ContainerBuilder().withName("alpine").withImage("alpine").withCommand("cat")
+                                        .withTty(true).build()) //
+                        .endSpec().build());
+                String podName = pod.getMetadata().getName();
+                LOGGER.log(Level.FINE, "Created test pod: {0}/{1}",
+                        new String[] { pod.getMetadata().getNamespace(), podName });
+                client.pods().withName(podName).waitUntilReady(30, TimeUnit.SECONDS);
+                client.pods().withName(podName).delete();
+
+                return FormValidation.ok("Connection test successful");
```

Comment:
Is this really just a "connection test" if we actually create a pod? The button also says "test connection"..

Reply:
any other suggestion?

Reply:
Good point 😄 .

I would only have stupid suggestions, write "test connection and creating pods" on the button, or remove the "creating pods" check.

Reply:
removed pod creation